From 403dd2ee396dbafc727fc7bf552ff41233811316 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Sun, 20 Sep 2026 00:04:04 +0200
Subject: [PATCH] avformat/dashdec: return EOF when a byte-range segment is
 exhausted

Fixes: Timeout
Fixes: t593bxSA0XOy
Found during triage/review of the security report q8954W25MJoa
Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>
---
 libavformat/dashdec.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/libavformat/dashdec.c b/libavformat/dashdec.c
index ffef5bfb2f3a..f3985788f40f 100644
--- a/libavformat/dashdec.c
+++ b/libavformat/dashdec.c
@@ -1747,8 +1747,11 @@ static int read_from_url(struct representation *pls, struct fragment *seg,
     int ret;
 
     /* limit read if the fragment was only a part of a file */
-    if (seg->size >= 0)
+    if (seg->size >= 0) {
         buf_size = FFMIN(buf_size, pls->cur_seg_size - pls->cur_seg_offset);
+        if (buf_size <= 0)
+            return AVERROR_EOF;
+    }
 
     ret = avio_read(pls->input, buf, buf_size);
     if (ret > 0)
